Refactor Config Service

acburdine opened the pull request:
closes #6266
- add "type" to valid keys in configuration api
- refactor ember config service to parse values based on provided type

TODO:
- [x] fix tests

acburdine commented:
@ErisDS I'm not sure if this is the best way to go about implementing the idea of "types" in the configuration, but it does fix the issue.

halfdan commented on core/client/app/services/config.js:
```diff
@@ -2,27 +2,24 @@ import Ember from 'ember';
 const {Service, _ProxyMixin, computed} = Ember;
 
-function isNumeric(num) {
+function _isNumeric(num) {
```
This is only used once and it's a one liner - just inline it where used?

kevinansfield replied:
I'm :+1: on it, it's following the same semantics as the other `isBlank`, `isEmpty`, etc helper methods and makes it easily searchable if it's ever needed elsewhere.

@acburdine are you able to change the name from `_isNumeric` to `isNumeric` please? The convention for the `_` prefix is to indicate that a method or instance variable should be considered private, helper functions don't really fall into that category.

halfdan replied:
@kevinansfield It literally just wraps the isNumeric from jQuery - so it's searchable. But :+1: for consistency with the other helpers.

kevinansfield replied:
Agreed that it's already searchable but we've had a previous issue where the `isNumeric` method content was different and was changed to the current implementation which is why I feel the explicit naming is preferable - should probably have mentioned that in my last reply :smile:

acburdine commented:
@kevinansfield fixed :smile:

kevinansfield commented:
@ErisDS I'm happy with the client-side changes here.

kevinansfield commented on core/client/app/services/config.js:
```diff
@@ -6,23 +6,20 @@ function isNumeric(num) {
-    } else if (isNumeric(val)) {
+    } else if (type === 'bool') {
+        return (val === 'true') ? true : false;
+    } else if (type === 'int' && _isNumeric(val)) {
```
I think you missed the rename here :smile:

acburdine committed:
refactor admin config to include explicit value types

closes #6266
- add "type" to valid keys in configuration api
- refactor ember config service to parse values based on provided type
